src/python/google_cloud_utils/storage.py

```diff
@@ -188,14 +188,27 @@ def list_blobs(self, remote_path):
     bucket = client.bucket(bucket_name)
     properties = {}
 
-    for blob in bucket.list_blobs(prefix=path):
+    if recursive:
+      delimiter = None
+    else:
+      delimiter = '/'
+
+    iterator = bucket.list_blobs(prefix=path, delimiter=delimiter)
+    for blob in iterator:
```

Comment:
Is it safe to rely on this behavior regarding delimiter presence? Maybe adding an extra check for `if recursive` wouldn't hurt?